src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt

```diff
@@ -2581,8 +2581,6 @@ namespace NServiceBus.Routing.MessageDrivenSubscriptions
     {
         public Publishers() { }
         public void Add(System.Type eventType, string publisher) { }
-        public void Add(System.Reflection.Assembly eventAssembly, string publisher) { }
```

Comment:
> So we explicitely don't want these APIs?

the configuration is still possible, but only on the `EndpointConfiguration` based API which still accepts an assembly + namespace. So the advanced API forces you to do stuff like assembly scanning yourself if you really want to use that.

> I assume the reasoning is that what happens if we get passed in an assembly which is not scanned by the assembly scanner, is that correct?

The main reason was that you call this API even at configuration time (if you know) why and it therefore can't internally rely on the `knownMessageTypes` which we currently rely on to avoid Assembly.GetType. If we allow specifying assemblies here, this will force a temporal coupling of this class because it can only be used after conventions and assembly scanning took place, while the actual routingtable doesn't care about these things, since it's a pure type based lookup table.

> Because it theory we could still support those APIs by essentially union in that assembly in the scanning mechanism.

that sounds somehow wrong to me. Why do you want to include assemblies again although you explicitly excluded them from assembly scanning (I think we only allow blacklisting for assembly scanning in v6)?